webpack.config.js

```diff
@@ -14,18 +25,407 @@ for (var i = 0; i < baseViewPaths.length; i++) {
+wax.on(Handlebars);
+wax.setLayoutPath(path.resolve(flavorBasePath, "templates"));
+
+
+// (3) Convert the config yaml to json
+// -----------------------------------------------------------------------------
+
+var config;
+var d = time(
+  () => { 
+    config = yaml.safeLoad(fs.readFileSync(flavorConfigPath)); 
+  }
+);
+log("Finished YAML parse", d);
+
+
+// (4) Compile base.hbs and index.html templates for the current flavor
+// -----------------------------------------------------------------------------
+
+var source, template;
+var d = time(
+  () => {
+    source = fs.readFileSync(
+      path.resolve(
+        flavorBasePath, 
+        "templates/index.html"
+      ), 
+      "utf8"
+    );
+    template = Handlebars.compile(source);
+  }
+);
+
+
+// (5) Localize the config for each language for the current flavor, precompile
+//     localized jstemplates Handlebars templates, and inject all localized
+//     content into the index-xx.html file
+// -----------------------------------------------------------------------------
+
+fs.readdirSync(flavorLocaleDir).forEach((langDir) => {
+
+  // Quick and dirty config clone
+  var thisConfig =  JSON.parse(JSON.stringify(config));
+
+  // Localize the config for the current language
+  // TODO: catenate flavor .po with base project .po!
+  var input = fs.readFileSync(
+    path.resolve(
+      flavorLocaleDir, 
+      langDir, 
+      "LC_MESSAGES", 
+      PO_FILE_NAME
+    )
+  );
+
+
+  var po = gettextParser.po.parse(input);
+  gt.addTranslations(langDir, "messages", po);
+  gt.setTextDomain("messages");
+  gt.setLocale(langDir);
+
+  var d = time(
+    () => {
+      walk(thisConfig, (val, prop, obj) => {
+        if (typeof val === "string") {
+          if (CONFIG_GETTEXT_REGEX.test(val)) {
+            val = val
+              .replace(CONFIG_GETTEXT_REGEX, "")
+              .replace(/\)$/, "");
+          }
+
+          obj[prop] = gt.gettext(val);
+        }
+      });
+    }
+  );
+  log("Finished localizing config for " + langDir, d);
+
+  // Add dataset site urls
+  thisConfig["datasets"] = datasetSiteUrls;
+
+
+  // (5a) Copy all jstemplates and flavor pages to a working directory from 
+  //      which the templates can be localized and precompiled. Also resolve 
+  //      flavor jstemplates overrides at this step
+  // ---------------------------------------------------------------------------
+
+  var d = time(
+    () => {
+      fs.copySync(
+        baseJSTemplatesPath,
+        outputJSTemplatesPath
+      );
+
+      fs.copySync(
+        flavorJSTemplatesPath,
+        outputJSTemplatesPath
+      );
+
+      fs.copySync(
+        flavorPagesPath,
+        outputJSTemplatesPath
+      );
+    }
+  );
+  log("Finished copying jstemplates assets", d);
+
+  // Localize jstemplates
+  fs.readdirSync(outputJSTemplatesPath).forEach((template) => {
+    if (template.endsWith("html")) {
+      var templ = path.resolve(outputJSTemplatesPath, template);
```

Comment:
Can we use a different variable name than `templ` here? It's kinda confusing as we already have a `template` variable